File: DATN_GameVH/src/main/java/com/gamevh/service/CouponOwnerService.java
package com.gamevh.service;

import java.util.List;

import com.gamevh.entities.CouponOwner;

public interface CouponOwnerService {
	CouponOwner findCouponByAccount(String username, String couponcode);
	
	List<CouponOwner> findCouponByUsername(String username);
	
	CouponOwner update(CouponOwner couponOwner);
}

File: DATN_GameVH_FE/site/services/api/CouponOwnerService.js
app.factory('CouponOwnerService', function ($http) {
    var baseUrl = host + '/api/couponowner';

    return {
        getCoupon: function (username, couponcode) {
            return $http.get(baseUrl + '/getcoupon/' + username + "/" + couponcode);
        }
    };
});
